stdlib/2/__builtin__.pyi

```diff
@@ -1333,16 +1333,119 @@ def next(__i: Iterator[_T]) -> _T: ...
+        'a', 'a+', '+a', 'at', 'ta', 'at+', 'a+t', '+at', 'ta+', 't+a', '+ta',
+        'x', 'x+', '+x', 'xt', 'tx', 'xt+', 'x+t', '+xt', 'tx+', 't+x', '+tx',
+        'U', 'rU', 'Ur', 'rtU', 'rUt', 'Urt', 'trU', 'tUr', 'Utr',
+    ]
+    _OPEN_BINARY_MODE = Literal[
+        'rb', 'br', 'rb+', 'r+b', '+rb', 'br+', 'b+r', '+br',
+        'wb', 'bw', 'wb+', 'w+b', '+wb', 'bw+', 'b+w', '+bw',
+        'ab', 'ba', 'ab+', 'a+b', '+ab', 'ba+', 'b+a', '+ba',
+        'xb', 'bx', 'xb+', 'x+b', '+xb', 'bx+', 'b+x', '+bx',
+        'rbU', 'rUb', 'Urb', 'brU', 'bUr', 'Ubr',
+    ]
+else:
+    _OPEN_TEXT_MODE = Literal[
+        'r', 'r+', '+r', 'rt', 'tr', 'rt+', 'r+t', '+rt', 'tr+', 't+r', '+tr',
+        'w', 'w+', '+w', 'wt', 'tw', 'wt+', 'w+t', '+wt', 'tw+', 't+w', '+tw',
+        'a', 'a+', '+a', 'at', 'ta', 'at+', 'a+t', '+at', 'ta+', 't+a', '+ta',
+        'U', 'rU', 'Ur', 'rtU', 'rUt', 'Urt', 'trU', 'tUr', 'Utr',
+    ]
+    _OPEN_BINARY_MODE = Literal[
+        'rb', 'br', 'rb+', 'r+b', '+rb', 'br+', 'b+r', '+br',
+        'wb', 'bw', 'wb+', 'w+b', '+wb', 'bw+', 'b+w', '+bw',
+        'ab', 'ba', 'ab+', 'a+b', '+ab', 'ba+', 'b+a', '+ba',
+        'rbU', 'rUb', 'Urb', 'brU', 'bUr', 'Ubr',
+    ]
+
 if sys.version_info >= (3, 6):
-    def open(file: Union[str, bytes, int, _PathLike[Any]], mode: str = ..., buffering: int = ..., encoding: Optional[str] = ...,
-             errors: Optional[str] = ..., newline: Optional[str] = ..., closefd: bool = ...,
-             opener: Optional[Callable[[str, int], int]] = ...) -> IO[Any]: ...
+    # Changed in version 3.6: Support added to accept objects implementing os.PathLike.
+    _OPEN_FILE = Union[str, bytes, int, _PathLike[Any]]
+elif sys.version_info >= (3,):
+    _OPEN_FILE = Union[str, bytes, int]
+else:
+    _OPEN_FILE = Union[unicode, int]
+
+if sys.version_info >= (3, 3):
+    # Changed in version 3.3: The opener parameter was added.
+    @overload
+    def open(
+        file: _OPEN_FILE,
+        mode: _OPEN_TEXT_MODE = ...,
+        buffering: int = ...,
+        encoding: Optional[str] = ...,
+        errors: Optional[str] = ...,
+        newline: Optional[str] = ...,
+        closefd: bool = ...,
+        opener: Optional[Callable[[str, int], int]] = ...,
+    ) -> TextIO: ...
+    @overload
+    def open(
+        file: _OPEN_FILE,
+        mode: _OPEN_BINARY_MODE,
+        buffering: int = ...,
+        encoding: None = ...,
+        errors: None = ...,
+        newline: None = ...,
+        closefd: bool = ...,
+        opener: Optional[Callable[[str, int], int]] = ...,
+    ) -> BinaryIO: ...
+    @overload
+    def open(
+        file: _OPEN_FILE,
+        mode: str,
+        buffering: int = ...,
+        encoding: Optional[str] = ...,
+        errors: Optional[str] = ...,
+        newline: Optional[str] = ...,
+        closefd: bool = ...,
+        opener: Optional[Callable[[str, int], int]] = ...,
+    ) -> IO[Any]: ...
+
 elif sys.version_info >= (3,):
```

Comment:
I think this branch can be removed because we no longer support 3.0 through 3.3.